### ISSUE TITLE
Scrapy is incompatible with Twisted 16.7.0rc1

### ISSUE THREAD

**Issue by redapple** [scrapy/scrapy#2479]
See [Twisted 16.7.0rc1 release annoucement](https://twistedmatrix.com/pipermail/twisted-python/2016-December/030984.html) for new features and changes.

https://github.com/twisted/twisted/pull/624 seems to be causing trouble (may not be the only one).

https://github.com/scrapy/scrapy/issues/2461 is also happening with 16.7.0rc1

**Comment by glyph** [comment]
Hi there! The Twisted release ticket is here: https://twistedmatrix.com/trac/ticket/8963

The intention of our compatibility policy is to avoid breaking projects that depend on Twisted.  In this case, it looks like Scrapy is technically violating some interfaces within Twisted, but in a way which used to work.  We hope you'll fix your implementation to match the interface, but in the meanwhile, please feel free to comment on that ticket and note that Scrapy is broken by the release, and hopefully before the release we can fix it so that Scrapy users can upgrade.

**Comment by redapple** [comment]
FYI, @glyph has identified one of the problems Scrapy is facing with 16.7.0rc1 (thanks for the swift resolution!): see https://twistedmatrix.com/trac/ticket/8963#comment:10

We still need to fix https://github.com/scrapy/scrapy/issues/2461

**Comment by Unode** [comment]
Just ran `pip install scrapy` on a new virtualenv and got `Twisted 17.1.0`.
Quite a lot of HTTPS websites started to fail (e.g. google) with a twisted error `SSL23_GET_SERVER_HELLO - tlsv1 alert internal error`.

Part of the problem is that code introduced after #1794 makes use of an import in `core/downloader/tls.py`which is no longer valid.
They symbol `_maybeSetHostNameIndication` has been removed from [_sslverify.py](https://github.com/twisted/twisted/blob/twisted-17.1.0/src/twisted/internet/_sslverify.py)

Also to avoid these incompatibilities shouldn't scrapy on pypi be limited to known working versions of dependencies?

**Comment by redapple** [comment]
Hi Renato,
Can I ask you to open a separate issue with a traceback showing the import
error? Thanks.
This suggests TLS with sni is not properly tested with current test suite.


Le 13 févr. 2017 1:25 AM, "Renato Alves" <notifications@github.com> a
écrit :

Just ran pip install scrapy on a new virtualenv and got Twisted 17.1.0.
Quite a lot of HTTPS websites started to fail (e.g. google) with a twisted
error SSL23_GET_SERVER_HELLO - tlsv1 alert internal error.

Part of the problem is that code introduced after #1794
<https://github.com/scrapy/scrapy/pull/1794> makes use of an import in
core/downloader/tls.pywhich is no longer valid.
They symbol _maybeSetHostNameIndication has been removed from _sslverify.py
<https://github.com/twisted/twisted/blob/twisted-17.1.0/src/twisted/internet/_sslverify.py>

Also to avoid these incompatibilities shouldn't scrapy on pypi be limited
to known working versions of dependencies?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/scrapy/scrapy/issues/2479#issuecomment-279268723>, or mute
the thread

**Comment by redapple** [comment]
Fixed in https://github.com/scrapy/scrapy/pull/2558
